sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -117,7 +117,7 @@ case class FilterExec(condition: Expression, child: SparkPlan)
 
   override def output: Seq[Attribute] = {
     child.output.map { a =>
-      if (a.nullable && notNullAttributes.contains(a.exprId)) {
+      if (a.nullable && notNullPreds.exists(_.semanticEquals(a))) {
```

Comment:
notNullPreds is a list of IsNotNull expressions. I think this semanticEquals comparison will fail at all?

Reply:
yeah, this will fail all. This change is not good enough. I'll try another way.